### PR TITLE
feat(Date): add DatePickerInput component

### PR DESCRIPTION
**Pull request by sun-tea**
- [x] Feature
- [x] Fix
- [ ] Enhancement

## Brief
Fix Date Picker component display.
Add a new component DatePickerInput which provides an input displaying a date picker.

## Description
Date Picker now displays the month of the current date when controlled.
DatePickerInput displays a Date Picker that can be updated through a Text Input with minimal validation (a valid Luxon DateTime).

untitled

## Todo - Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.

### CODE REVIEW

File: src/Date/DatePickerInput/index.js [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+  componentWillUnmount() {
+    document.removeEventListener('mousedown', this.handleClick, false);
+  }
+
+  /**
+   * Toggles visibility of the date picker.
+   * Closes the date picker if the user has clicked outside this component
+   *
+   */
+  handleClick = e => {
```
Comment (spartDev) [inline src/Date/DatePickerInput/index.js]: You should put `@param` in description

Comment (spartDev) [reply src/Date/DatePickerInput/index.js]: Should should use `event` instead of `e`

File: src/Date/DatePickerInput/index.js [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+      this.setState({
+        isDatePickerOpen: false,
+      });
+    }
+  };
+
+  /**
+   * Handles change of date from DatePicker.
+   */
+  handleDateChange = dateValue => {
```
Comment (spartDev) [inline src/Date/DatePickerInput/index.js]: You should put `@param` in the description

File: src/Date/DatePickerInput/index.js [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+  /**
+   * Handles change of value from text input.
+   * If the value is valid from a Luxon point of view, it will be passed to the calendar
+   * Otherwise, we set the date value for the DatePicker to the current time.
+   * Thus we keep the text and date values out of sync to allow the user to update the date
+   * while keeping a valid date for the DatePicker
+   * Validation can be handled higher with the callback
+   *
+   */
+  handleInputChange = textValue => {
```
Comment (spartDev) [inline src/Date/DatePickerInput/index.js]: You should put `@param` in the description

**Comment by sun-tea** [comment]
@spartDev all done!

**Pull request merged**